tests/interfaces/test_jax_jit_qnode.py

```diff
@@ -1508,6 +1513,8 @@ def test_hamiltonian_expansion_analytic(
         are non-commuting groups and the number of shots is None
         and the first and second order gradients are correctly evaluated"""
         gradient_kwargs = {}
+        if dev.name == "reference.qubit":
+            pytest.xfail("reference.qubit does not support hamiltonians")
```

Comment:
Isn't it calling `split_non_commuting` in preprocessing? Shouldn't that enable hamiltonian support?

Reply:
Is it just the `assert_no_called` part that fails, or something else?

Reply:
The message was a bit misleading I guess, but the actual issue iirc was that the spying does not work on a transform. When `split_non_commuting` is replaced by the mocker, you can't add it to a transform program anymore, and preprocessing fails.